packages/form-js-viewer/src/render/components/form-fields/FilePicker.js

```diff
@@ -1,8 +1,86 @@
+import { formFieldClasses } from '../Util';
+import { Label } from '../Label';
+import { Errors } from '../Errors';
+import { useRef, useState } from 'preact/hooks';
+import { useSingleLineTemplateEvaluation } from '../../hooks';
+
+const type = 'filepicker';
+
 /**
+ * @typedef Props
+ * @property {(props: { value: string }) => void} onChange
+ * @property {string} domId
+ * @property {string[]} errors
+ * @property {boolean} disabled
+ * @property {boolean} readonly
+ * @property {boolean} required
+ * @property {Object} field
+ * @property {string} field.id
+ * @property {string} [field.label]
+ * @property {string} [field.accept]
+ * @property {boolean} [field.multiple]
+ *
+ * @param {Props} props
  * @returns {import("preact").JSX.Element}
  */
-export function FilePicker() {
-  return null;
+export function FilePicker(props) {
+  /** @type {import("preact/hooks").Ref<HTMLInputElement>} */
+  const fileInputRef = useRef(null);
+  /** @type {[File[],import("preact/hooks").StateUpdater<File[]>]} */
+  const [selectedFiles, setSelectedFiles] = useState([]);
+  const { field, onChange, domId, errors = [], disabled, readonly, required } = props;
+  const { label, multiple = '', accept = '', id } = field;
+  const evaluatedAccept = useSingleLineTemplateEvaluation(accept);
+  const evaluatedMultiple =
+    useSingleLineTemplateEvaluation(typeof multiple === 'string' ? multiple : multiple.toString()) === 'true';
+  const errorMessageId = `${domId}-error-message`;
+
+  return (
+    <div class={formFieldClasses(type, { errors, disabled, readonly })}>
+      <Label htmlFor={domId} label={label} required={required} />
+      <input
+        type="file"
+        className="fjs-hidden"
+        ref={fileInputRef}
+        id={domId}
+        name={domId}
+        multiple={evaluatedMultiple === false ? undefined : evaluatedMultiple}
+        accept={evaluatedAccept === '' ? undefined : evaluatedAccept}
```

Comment:
One issue I see here is that this accept does not enforce the file type. You can still navigate to 'All files' and it'll accept it in the upload. Something that will probably need doing, although not crucial right now.

Reply:
Yeah, but this is on the OS level. I researched it and it looks like we have no control over that